### PR TITLE
Queue Implementation

### PR DESCRIPTION
**Pull request by ishrutik**
<hr>

## Description 📜

Here queue is implemented using array and linked list.

<hr>

## Type of contribution 📝

- [x] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/#CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by ChetasShree** [comment]
@ishrutik  Add a relative link to the readme file where u have placed ur content. Here is how you add a
 [relative link] (https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file)

**Comment by ishrutik** [comment]
> @ishrutik Add a relative link to the readme file where u have placed ur content. Here is how you add a [relative link] (https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file)

Done @ChetasShree

**Comment by ChetasShree** [comment]
Not in ur readme file, when u open the main folder of C_CPP->Data Structures->Standard Template Library file, inside that,u can see a readme file ,where all the relative links are there, in that readme file you have to add it.

Plus have you created ur own STL folder? if yes, then pls don't, add it in the folder which is already there.

**Comment by ishrutik** [comment]
> Not in ur readme file, when u open the main folder of C_CPP->Data Structures->Standard Template Library file, inside that,u can see a readme file ,where all the relative links are there, in that readme file you have to add it.
> 
> Plus have you created ur own STL folder? if yes, then pls don't, add it in the folder which is already there.

Done @ChetasShree

File: C_CPP/Data Structures/Queue/readme.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
-- [Queue](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/readme.md)<br>
-- [Queue using Arrays](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/Queue_Using_Array.cpp)<br>
-- [Queue using linked list](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/Queue_Using_LL.cpp)
+- [Queue](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/readme.md)<br>
```
Comment (ChetasShree) [inline C_CPP/Data Structures/Queue/readme.md:1]: this file is empty

File: C_CPP/Data Structures/Queue/readme.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
...
+- [Queue using Arrays](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/Queue_Using_Array.cpp)<br>
+- [Queue using linked list](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/Queue_Using_LL.cpp)
+- [Implementation of queue using stack](./implementation_of_queue_using_stack.md)
```
Comment (ChetasShree) [inline C_CPP/Data Structures/Queue/readme.md:4]: Pls use the proper format of md file

**Review by ChetasShree — CHANGES_REQUESTED** [review]

File: C_CPP/Data Structures/Queue/readme.md (hunk 1 above)
Comment (ishrutik) [reply C_CPP/Data Structures/Queue/readme.md:1]: > this file is empty

@ChetasShree readme updated

File: C_CPP/Data Structures/Queue/readme.md (hunk 2 above)
Comment (ishrutik) [reply C_CPP/Data Structures/Queue/readme.md:4]: > Pls use the proper format of md file
@ChetasShree [Implementation of queue using stack](./implementation_of_queue_using_stack.md) this is already merged file. This is not mine

File: C_CPP/Data Structures/Queue/Queue Implementation/readme.md [hunk 3]
```diff
@@ -1 +1,38 @@
+# QUEUE
+
+Queue is a linear data structure which follows a particular order in which operations are performed i.e. **First in First Out** also known as **FIFO**.
+FIFO principle implies that the first element inserted will be the first one to be removed.<br>
+Real life example of queue can be,line of vehicles on road during heavy traffic.Here the vehicle which came first will be the one to pass the traffic.<br><br>
+<img src="https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/Queue_Real_Life_Example.jpg"/>
+<br><br><br>
+Queue can be implemented using both linked list and arrays.<br><br>
+Implementation using Array<br>
+[Queue Implementation using Array](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Standard%20Template%20Library%20(Vectors%2C%20Maps%20etc)/Queue%20Implementation/Queue_Using_Array.cpp)
```
Comment (ChetasShree) [inline C_CPP/Data Structures/Queue/Queue Implementation/readme.md:10]: it says page not found

File: C_CPP/Data Structures/Queue/Queue Implementation/readme.md [hunk 4]
```diff
@@ -1 +1,38 @@
...
+FIFO principle implies that the first element inserted will be the first one to be removed.<br>
+Real life example of queue can be,line of vehicles on road during heavy traffic.Here the vehicle which came first will be the one to pass the traffic.<br><br>
+<img src="https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/Queue_Real_Life_Example.jpg"/>
+<br><br><br>
+Queue can be implemented using both linked list and arrays.<br><br>
+Implementation using Array<br>
+[Queue Implementation using Array](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Standard%20Template%20Library%20(Vectors%2C%20Maps%20etc)/Queue%20Implementation/Queue_Using_Array.cpp)
+<br><br>
+Implementation using Linked List<br>
+[Queue Implementation using Linked List](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Standard%20Template%20Library%20(Vectors%2C%20Maps%20etc)/Queue%20Implementation/Queue_Using_LL.cpp)
```
Comment (ChetasShree) [inline C_CPP/Data Structures/Queue/Queue Implementation/readme.md:13]: It says page not found

**Review by ChetasShree — CHANGES_REQUESTED** [review]

**Comment by ChetasShree** [comment]
Please mention an index line in readme along with relative link as well, and the rest is good to go

**Comment by ishrutik** [comment]
> Please mention an index line in readme along with relative link as well, and the rest is good to go

@ChetasShree I didn't understand ma'am ,,index line??

**Comment by ishrutik** [comment]
> > Please mention an index line in readme along with relative link as well, and the rest is good to go
> 
> @ChetasShree I didn't understand ma'am ,,index line??

@ChetasShree in main readme of queue,I added links of my readme and code  already

**Comment by ChetasShree** [comment]
Go to C_CPP->Data Structures->Queue ,in that you can see a readme file ryt? over there u have to add a relative link of ur file

**Comment by ishrutik** [comment]
> Go to C_CPP->Data Structures->Queue ,in that you can see a readme file ryt? over there u have to add a relative link of ur file.
> 
> Add a relative link to the readme file where u have placed ur content. Here is how you add a [relative link] (https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-

> Go to C_CPP->Data Structures->Queue ,in that you can see a readme file ryt? over there u have to add a relative link of ur file.
> 
> Add a relative link to the readme file where u have placed ur content. Here is how you add a [relative link] (https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file)

@ChetasShree i did this already, kindly check

**Comment by ishrutik** [comment]
> Go to C_CPP->Data Structures->Queue ,in that you can see a readme file ryt? over there u have to add a relative link of ur file.
> 
> Add a relative link to the readme file where u have placed ur content. Here is how you add a [relative link] (https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file)

@ChetasShree I did this already, kindly check and merge

**Comment by ChetasShree** [comment]
Amazing work @ishrutik  !!!
@Suvraneel @Saviour1001 pls review this PR and merge it :)
Thank you !!

**Pull request merged**